### PR TITLE
Refactor tables demo launch file to easily include other environments

### LINKED ISSUE #7: cic_tables_spawn_sim_objects.launch is missing
#5 contains references to the following file:

```text
mobipick_gazebo/launch/worlds/cic_tables_spawn_sim_objects.launch
```

However, this file is still missing.

### PR DESCRIPTION
**Pull request by oscar-lima**
Refactor the mobipick launch configuration to allow other repository to derive from it.

### CODE REVIEW

**Review by mintar — APPROVED** [review]
LGTM! The following files are still missing, but I assume they will be added in a different PR:

```text
mobipick_gazebo/launch/worlds/cic_tables_spawn_sim_objects.launch
mobipick_gazebo/launch/worlds/truck_assembly_spawn_sim_objects.launch
```

**Comment by mintar** [comment]
By the way, I've rebased this branch onto the latest noetic branch and force-pushed.

**Comment by mintar** [comment]
@oscar-lima: I believe 0d094c5 adds the wrong file. #7 is about the `cic_tables` file. The `truck_assembly` file will be added in #6.

**Review by mintar — CHANGES_REQUESTED** [review]
@oscar-lima: I believe 0d094c5 adds the wrong file. #7 is about the `cic_tables` file. The `truck_assembly` file will be added in #6.

**Comment by oscar-lima** [comment]
> @oscar-lima: I believe [0d094c5](https://github.com/DFKI-NI/mobipick/commit/0d094c57935847397dcb28cccbe79ef59b7dc3b8) adds the wrong file. #7 is about the `cic_tables` file. The `truck_assembly` file will be added in #6.

reverted in https://github.com/DFKI-NI/mobipick/commit/20d2c5174bd5dae60040f3dbbb2ea17ef3b2bc5e

**Comment by oscar-lima** [comment]
> LGTM! The following files are still missing, but I assume they will be added in a different PR:
> 
> ```
> mobipick_gazebo/launch/worlds/cic_tables_spawn_sim_objects.launch
> mobipick_gazebo/launch/worlds/truck_assembly_spawn_sim_objects.launch
> ```

Added in https://github.com/DFKI-NI/mobipick/commit/3b0f03cceaa9a52a0d872e46507fedeb4787f4ef

File: mobipick_gazebo/launch/mobipick/tables_demo.launch [hunk 1]
```diff
@@ -4,6 +4,9 @@
   <arg name="tf_prefix" default="$(arg namespace)" doc="tf_prefix to be used" />
   <arg name="robot_name" default="mobipick" doc="Sets the name of the robot in gazebo" />
 
+  <arg name="world" default="pbr_moelk"/>
```
Comment (jarkenau) [inline mobipick_gazebo/launch/mobipick/tables_demo.launch:7]:
```suggestion
  <arg name="world" default="pbr_moelk" doc="world of the tables demo, options: pbr_moelk, pbr_cic" />
```

Comment (jarkenau) [inline mobipick_gazebo/launch/mobipick/tables_demo.launch]: What I currently find unintuitive, is that options are provided for the `world config` but not for the `world` argument itself and consequently also the robot pose. This makes the impression that you can change the `world_config` to one of the three values and everything will work.

**Review by jarkenau — COMMENTED** [review]
Cleaning up the history by squashing some commits would also be appreciated.

**Comment by jarkenau** [comment]
Closes #7

File: mobipick_gazebo/launch/mobipick/tables_demo.launch (hunk 1 above)
Comment (oscar-lima) [reply mobipick_gazebo/launch/mobipick/tables_demo.launch]: added documentation in https://github.com/DFKI-NI/mobipick/commit/2fbb7af21533511051f4d1600f5cb245566b9305 and https://github.com/DFKI-NI/mobipick_labs/commit/b878830224d4c8450a2c350115ac60c944d25dee , does this solve the issue?

**Comment by oscar-lima** [comment]
> Cleaning up the history by squashing some commits would also be appreciated.

we can click "squash and merge" when we merge the pull request

**Comment by mintar** [comment]
I just tried running everything. Here are the results:

## WORKING

```
roslaunch tables_demo_bringup demo_sim.launch world:=pbr_moelk world_config:=moelk_tables
```

## NOT WORKING

```
$ roslaunch tables_demo_bringup demo_sim.launch world:=pbr_cic world_config:=truck_assembly
RLException: error loading <rosparam> tag: 
	file does not exist [/home/mguenther/ros/noetic/mobipick/src/mobipick_labs/tables_demo_bringup/config/truck_assembly_planning_scene.yaml]
XML is <rosparam file="$(arg planning_scene_config)" command="load"/>
```

```
$ roslaunch tables_demo_bringup demo_sim.launch world:=pbr_cic world_config:=cic_tables    
RLException: while processing /home/mguenther/ros/noetic/mobipick/src/mobipick_labs/tables_demo_bringup/launch/includes/navigation/cic_tables.launch:
Invalid roslaunch XML syntax: [Errno 2] No such file or directory: '/home/mguenther/ros/noetic/mobipick/src/mobipick_labs/tables_demo_bringup/launch/includes/navigation/cic_tables.launch'
```

Am I doing something wrong?

**Comment by oscar-lima** [comment]
> Am I doing something wrong?

No, the file is not pushed yet. This PR was supposed to be a refactor of the tables demo and somehow grew more than expected. Sorry about that, would it be ok if I make a separate PR to include it? or do we really want a perfectly clean git history.

**Comment by mintar** [comment]
A separate PR would be best!

**Pull request merged**